code/datums/helper_datums/butchering.dm

```diff
@@ -349,3 +352,249 @@ var/global/list/animal_butchering_products = list(
+			if(!istype(H.gloves))
+				butchSpeed += 0.25
+				if(!toolName)
+					toolName = "claws"
+		if(isgrue(H))
+			toolName = "grue"
+			butchSpeed += 0.5
+	else
+		butchSpeed = 0.5
+	if(!butchSpeed)
+		return
+	return list(BUTCHER_SPEED = butchSpeed, BUTCHER_TOOL = theTool, BUTCHER_TNAME = toolName)
+
+
+/mob/living/proc/butcherMenuStep(mob/user)
+	var/list/butcherType = list()
+	if(meat_type && meat_amount > meat_taken)
+		butcherType += "Butcher"
+	for(var/datum/butchering_product/BP in butchering_drops)
+		if(BP.amount)
+			butcherType += BP.verb_name
+	if(!butcherType.len)
+		to_chat(user, "<span class='notice'>There's nothing to butcher.</span>")
+		return
+	butcherType += "Cancel"
+	return butcherType
+
+
+/mob/living/proc/butcherChooseStep(mob/user, butcherOptions, butcherTool)
+	var/choice = input(user,"What would you like to do with \the [src]?","Butchering") in butcherOptions
+	if(!butcherCheck(user, butcherTool))
+		return 0
+	if(choice == "Cancel")
+		return 0
+	if(choice == "Butcher")
+		return BUTCHER_MEAT
+	if(!choice || !butchering_drops.len)
+		return 0
+	var/theProduct = getProduct(choice)
+	return theProduct
+
+
+/mob/living/proc/getProduct(choice)
+	for(var/datum/butchering_product/BP in butchering_drops)
+		if(BP.verb_name == choice)
+			return BP
+
+
+/mob/living/proc/butcherCarveStep(mob/user, speed_mod, datum/butchering_product/typeOfCarve, toolName, var/butcherTime = 20)
+	var/butcherWord = null
+	if(typeOfCarve == BUTCHER_MEAT)
+		butcherWord = "butchering"
+		butcherTime *= size
+		if(meat_taken >= meat_amount)
+			to_chat(user, "<span class='info'>There's no meat left!.</span>")
+			return
+	else
+		butcherWord = typeOfCarve.verb_gerund
+		butcherTime = typeOfCarve.butcher_time
+	if(!butcherWord)
+		return FALSE
+	user.visible_message("<span class='notice'>[user] starts [butcherWord] \the [src][toolName ? " with \the [toolName]" : ""].</span>",\
+		"<span class='info'>You start [butcherWord] \the [src].</span>")
+	being_butchered = TRUE
+	if(!do_after(user, src, butcherTime / speed_mod))
+		to_chat(user, "<span class='warning'>You stop [butcherWord] \the [src].</span>")
+		being_butchered = FALSE
+		return FALSE
+	being_butchered = FALSE
+	to_chat(user, "<span class='info'>You finish [butcherWord] \the [src].</span>")
+	return TRUE
+
+
+/mob/living/proc/butcherMeat(mob/user, var/tool_name)
+	var/theMeat = drop_meat(loc)
+	if(theMeat)
+		if(tool_name)
+			if(!advanced_butchery)
+				advanced_butchery = new()
+			advanced_butchery.Add(tool_name)
+		modMeat(user, theMeat)
+	meatEndStep(user)
+
+
+/mob/living/proc/drop_meat(location)
+	if(!meat_type)
+		return
+	if(meat_taken >= meat_amount)
+		return
+	meat_taken++
+	if(!ispath(meat_type, /obj/item/weapon/reagent_containers/food/snacks/meat))	//For 2 years sheet/bone thought "skeleton" was a number
+		var/NM = new meat_type(location)
+		return NM
+	var/obj/item/weapon/reagent_containers/food/snacks/meat/M = null
+	if(ishuman(src))
```

Comment:
This check and the one below for `simple_animal` could be delegated to implementations of `drop_meat` for those types instead

Reply:
For simple animal you're right and I should have thought of that, but for humans due to how meat has a special snowflake New() I think the most efficient way is to specify there since anything else would just be a similar check.

@DrSnips Ya that's why I made it silly, but I guess I should probably specify a bit more

Reply:
Typechecking `src` (which this is) is always categorically a no-no.